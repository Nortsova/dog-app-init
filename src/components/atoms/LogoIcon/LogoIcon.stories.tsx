import type { Meta, StoryObj } from '@storybook/react';
import { LogoIcon } from '.';

const meta = {
  title: 'Atoms/LogoIcon',
  component: LogoIcon,
  argTypes: {
    variant: {
      options: ['default'],
      control: 'radio'
    }
  }
} satisfies Meta<typeof LogoIcon>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    variant: 'default'
  }
};
