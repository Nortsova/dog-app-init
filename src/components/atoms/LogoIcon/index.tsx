import { SvgIcon, Link as LinkLogo, styled } from '@mui/material';

import IconLogo from '../../../assets/icons/LogoIcon.svg?react';

const Link = styled(LinkLogo)({
  underline: 'none'
});

export const LogoIcon = () => (
  <Link href="/">
    <SvgIcon
      component={IconLogo}
      sx={{ width: 201, height: 40 }}
      inheritViewBox
    />
  </Link>
);
