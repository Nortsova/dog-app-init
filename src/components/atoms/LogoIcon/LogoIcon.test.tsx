import { composeStories } from '@storybook/react';
import * as stories from './LogoIcon.stories';
import { testSnapshot } from '../../../test/utils/snapshot.utils';

describe('LogoIcon', () => {
  const { Primary } = composeStories(stories);
  testSnapshot(<Primary />);
});
