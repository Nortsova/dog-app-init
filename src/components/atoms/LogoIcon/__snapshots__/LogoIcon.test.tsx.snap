// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoIcon should match snapshot Component 1`] = `
.emotion-0 {
  margin: 0;
  font: inherit;
  color: #921FED;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-decoration-color: rgba(146, 31, 237, 0.4);
  underline: none;
}

.emotion-0:hover {
  text-decoration-color: inherit;
}

<div>
  <a
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways emotion-0"
    href="/"
  >
    <test-file-stub
      aria-hidden="true"
      classname="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gg0jzq-MuiSvgIcon-root"
      focusable="false"
    />
  </a>
</div>
`;
